### PR TITLE
Refactor Dockerfile/main.go/config

### PR DESCRIPTION
**Pull request by chiragk25**

### CODE REVIEW

**Comment by chiragk25** [comment]
This change -

- Moves all main.go files under `cmd/` and use a single go.mod file for the project
- Moves all Dockerfiles under root directory
- Moves all Kubernetes config files under `config/k8s/`
- Moves all scripts under `scripts/`

File: Dockerfile.tlsreload [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# syntax=docker/dockerfile:1
+
+FROM golang:1.21.3-alpine AS builder
+RUN mkdir -p src/github.com/rturner3/spire-mysql-demo
+WORKDIR /go/src/github.com/rturner3/spire-mysql-demo
+COPY go.mod go.sum .
+RUN go mod download
+COPY cmd cmd
+RUN go build -o /tls-bootstrap ./cmd/mysql/tls-reload
```
Comment (rturner3) [inline Dockerfile.tlsreload]:
```suggestion
RUN go build -o /tls-reload ./cmd/mysql/tls-reload
```

**Review by rturner3 — APPROVED** [review]

**Review by rturner3 — CHANGES_REQUESTED** [review]
Oops, didn't mean to approve. Noticed one bug in one of the Dockerfiles.

**Comment by chiragk25** [comment]
Verified `./scripts/push-mysql-tls-images.sh` is able to push both `tls-reload` and `tls-bootstrap` images

**Review by rturner3 — APPROVED** [review]

**Pull request merged**